packages/waku/src/lib/middleware/dev-server-impl.ts

```diff
@@ -460,6 +460,8 @@ export const devServer: Middleware = (options) => {
     const body = await readablePromise;
     if (body) {
       ctx.res.body = body;
+    } else if (ctx.res.status === 404) {
+      delete ctx.res.status;
```

Comment:
I'm guessing this was the main fix? When the body and status are set, it finalizes the hono request, but by deleting the status here, you let it pass through to the Hono error handler.